code.js

```diff
@@ -1,3 +1,55 @@
+// Global variable for the counting makes tracking easier
+// Pretty sure globals need to be var
+var numPerms = 0;
```

Comment:
This will break if you run the function multiple times.

Reply:
You're right, I see, I'm not at my laptop right now I'll fix it soon